corehq/apps/hqcase/tests/test_case_update_api.py

```diff
@@ -490,6 +507,52 @@ def test_index_reference_to_uncreated_external_id(self):
         child = CommCareCase.objects.get_case_by_external_id(self.domain, 'harmon-luchenko')
         self.assertEqual(parent.case_id, child.get_index('parent').referenced_id)
 
+    def test_update_by_external_id(self):
+        case = self._make_case()
+        res = self.client.put(
+            reverse('case_api', args=(self.domain,)),
+            {
+                'external_id': case.external_id,
+                'properties': {'champion': 'true'},
+            },
+            content_type="application/json;charset=utf-8",
+        )
+        self.assertEqual(res.status_code, 200)
+        case = CommCareCase.objects.get_case(case.case_id, self.domain)
+        self.assertEqual(case.dynamic_case_properties().get('champion'), 'true')
+
+    def test_update_by_external_id_doesnt_exist(self):
+        res = self.client.put(
+            reverse('case_api', args=(self.domain,)),
+            {
+                'external_id': 'notarealcaseid',
+                'properties': {'champion': 'true'},
+            },
+            content_type="application/json;charset=utf-8",
+        )
+        self.assertEqual(res.status_code, 400)
+        self.assertEqual(res.json()['error'], "Could not find a case with external_id 'notarealcaseid'")
+
+    def test_bulk_update_by_external_id(self):
+        case = self._make_case()
+        res = self._bulk_update_cases([{
+            'create': False,
+            'external_id': case.external_id,
+            'owner_id': 'us_chess_federation',
+        }])
+        self.assertEqual(res.status_code, 200)
+        case = CommCareCase.objects.get_case(case.case_id, self.domain)
+        self.assertEqual(case.owner_id, 'us_chess_federation')
+
+    def test_bulk_update_external_id_doesnt_exist(self):
+        res = self._bulk_update_cases([{
+            'create': False,
+            'external_id': 'notarealcaseid',
+            'owner_id': 'us_chess_federation',
+        }])
+        self.assertEqual(res.status_code, 400)
+        self.assertEqual(res.json()['error'], "Could not find a case with external_id 'notarealcaseid'")
+
```

Comment:
could do with a test for creating / updating an index using `external_id`

Reply:
:+1: I'll do that on the child PR

Reply:
oh, actually there already is one:
https://github.com/dimagi/commcare-hq/blob/4baad8c17b037a0d6a7e475185e421a18977ff2a/corehq/apps/hqcase/tests/test_case_update_api.py#L298-L315